### PR TITLE
DOC: extend the section on adding metadata to plans

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

File: doc/source/metadata.rst [hunk 1]
```diff
@@ -1,22 +1,84 @@
...
+There are somethings that we know a-priori before doing an experiment;
+where are we? who is the user? what sample are we looking at? what did
+the user just ask us to do?  These are all things that we can, in
+principle, know independent of the control system.  These are the
+prime candidates for inclusion in the `Start Document
+<https://blueskyproject.io/event-model/data-model.html#run-start-document>`_.
+Downstream DataBroker provides tools to do rich searches on this data.
+The more information you can include the better.
+
+There is some information that we need to is nominally independent of
```
Comment (mrakitin) [inline doc/source/metadata.rst]:
```suggestion
There is some information that we need to that is nominally independent of
```

File: doc/source/metadata.rst [hunk 2]
```diff
@@ -1,22 +1,84 @@
...
+any particular device but we need to consult the controls system
+about.  For example the location of important, but un-scanned motors
+or the configuration of filters.  If the values *should* be fixed over
```
Comment (mrakitin) [inline doc/source/metadata.rst]:
```suggestion
or the configuration of beam attenuators.  If the values *should* be fixed over
```

File: doc/source/metadata.rst [hunk 3]
```diff
@@ -1,22 +1,84 @@
...
+the course of the experiment then this it is a good candidate for
+being a "baseline device" either via the `Supplemental pre-processor
+<https://blueskyproject.io/bluesky/tutorial.html#baseline-readings-and-other-supplemental-data>`_
+or explicitly in custom plans.  This will put the readings in a separate stream
+(which is a peer to the "primary" data).  In principle, these values could be
+read from the control system once and put into the Start document along with
+the a-priori information, however that has several draw backs:
+
+1. There is only ever 1 reading of the values so if they do drift during
+   data acquisition you will never know
```
Comment (mrakitin) [inline doc/source/metadata.rst]:
```suggestion
   data acquisition, you will never know.
```

File: doc/source/metadata.rst [hunk 4]
```diff
@@ -1,22 +1,84 @@
...
+2. We can not automatically capture information about the device like
+   we do for data in Events.  This include things like the datatype,
```
Comment (mrakitin) [inline doc/source/metadata.rst]:
```suggestion
   we do for data in Events.  This includes things like the datatype,
```

File: doc/source/metadata.rst [hunk 5]
```diff
@@ -1,22 +1,84 @@
...
+   units, and shape of the value and any configuration information the hardware
```
Comment (mrakitin) [inline doc/source/metadata.rst]:
```suggestion
   units, and shape of the value and any configuration information about the hardware
```

File: doc/source/metadata.rst [hunk 6]
```diff
@@ -1,22 +1,84 @@
...
+   it is being read from.
+
+
+A third class of information that can be called "metadata" is
+configuration information of a pieces of hardware.  This is things
```
Comment (mrakitin) [inline doc/source/metadata.rst]:
```suggestion
configuration information of pieces of hardware.  This is things
```

File: doc/source/metadata.rst [hunk 7]
```diff
@@ -1,22 +1,84 @@
...
+like the velocity of a motor or the integration time of a detector.
+These readings are embeded in the `Descriptor
```
Comment (mrakitin) [inline doc/source/metadata.rst]:
```suggestion
These readings are embedded in the `Descriptor
```

File: doc/source/metadata.rst [hunk 8]
```diff
@@ -1,22 +1,84 @@
...
+<https://blueskyproject.io/event-model/data-model.html#event-descriptor>`_
+and are extracted from the hardware via the `read_configuration
+<https://blueskyproject.io/bluesky/hardware.html#ReadableDevice.read_configuration>`_
+method the hardware.  We expect that these values will not change over
```
Comment (mrakitin) [inline doc/source/metadata.rst]:
```suggestion
method of the hardware.  We expect that these values will not change over
```

File: doc/source/metadata.rst [hunk 9]
```diff
@@ -1,22 +1,84 @@
...
+should be treated as "data".  Either as an explicit part of the
+experimental plan or via :ref:`async_monitoring`.
+
+
+Adding to the Start Document
+============================
+
+
+When the RunEngine mints a Start document it includes structured data.  That
+information can be inject in via several mechanisms:
```
Comment (mrakitin) [inline doc/source/metadata.rst]:
```suggestion
information can be injected in via several mechanisms:
```

File: doc/source/metadata.rst [hunk 10]
```diff
@@ -130,6 +192,24 @@ If there is a conflict, ``RE`` keywords takes precedence. So
 would override the individual 'purpose' metadata from the plan, marking all
 three as purpose=test.
 
+If you define your own plans it is best practice have them take a keyword only
```
Comment (mrakitin) [inline doc/source/metadata.rst]:
```suggestion
If you define your own plans, it is best practice have them take a keyword only
```

File: doc/source/metadata.rst [hunk 11]
```diff
@@ -227,6 +307,19 @@ delete a key you want to stop using,
...
+
+
+   In general we recommend against putting device readings in the Start
+   document, however it is critical that you do not put device readings
+   in this dictionary.  The value will remain until you change it and not
+   track the state of the hardware.  This will result in recording incorrect
+   data!
+
+   This can be particularly dangerous if ``RE.md`` is backed by a
+   persistent data store (see next section)
```
Comment (mrakitin) [inline doc/source/metadata.rst]:
```suggestion
   persistent data store (see next section).
```

**Comment by danielballan** [comment]
This is great!

**Pull request merged**